R/pkg/R/catalog.R

```diff
@@ -403,21 +406,95 @@ listTables <- function(databaseName = NULL) {
   dataFrame(callJMethod(jdst, "toDF"))
 }
 
+#' Checks if the table with the specified name exists.
+#'
+#' Checks if the table with the specified name exists.
+#'
+#' @param tableName name of the table, allowed to be qualified with catalog name
+#' @rdname tableExists
+#' @name tableExists
+#' @examples
+#' \dontrun{
+#' sparkR.session()
+#' databaseExists("spark_catalog.default.myTable")
+#' }
+#' @note since 3.4.0
+tableExists <- function(tableName) {
+  sparkSession <- getSparkSession()
+  if (class(tableName) != "character") {
+    stop("tableName must be a string.")
+  }
+  catalog <- callJMethod(sparkSession, "catalog")
+  callJMethod(catalog, "tableExists", tableName)
+}
+
+#' Get the table with the specified name
+#'
+#' Get the table with the specified name
+#'
+#' @param tableName the qualified or unqualified name that designates a table, allowed to be
+#'                  qualified with catalog name
+#' @return A data.frame.
```

Comment:
creating a new struct/class in R seems complicated, here just return a `data.frame`

Reply:
```r
> 
> schema <- structType(structField("name", "string"), structField("age", "integer"), structField("height", "float"))
> createTable("people", source = "json", schema = schema)
SparkDataFrame[name:string, age:int, height:float]
> tbl <- getTable("people")
> tbl
    name       catalog namespace description tableType isTemporary
1 people spark_catalog   default        NULL   MANAGED       FALSE
> 
```

Reply:
alternatively, you can use named list (e.g., `list(name=...)`) or `env` instance.

Reply:
probably using named list is a better idea given that we use namedtuple in Python side.

Reply:
named list seems a better choice, let me have a try